smartseq2_single_sample/run_multi_samples/scripts/calculate_cost_workflow.py

```diff
@@ -0,0 +1,255 @@
+
+            if y.startswith("ok"):
+                end = dateutil.parser.parse(x['endTime'])
+                found_end = True
+
+    # if we are preempted or if cromwell used previously cached results, we don't even get a start time from JES
+    # use the Cromwell start time which is earlier but not wrong
+    if not found_start and (was_preempted(call_info) or used_cached_results(call_info)):
+        start = dateutil.parser.parse(call_info['start'])
+        found_start = True
+
+    # if we are preempted or if cromwell used previously cached results, we don't get an endtime from JES right now...
+    # use the Cromwell end time which is later but not wrong
+    if not found_end and (was_preempted(call_info) or used_cached_results(call_info)):
+        end = dateutil.parser.parse(call_info['end'])
+        found_end = True
+
+    if not found_start or not found_end:
+        exit("Unable to find start or end", call_info)
+
+    # round up to nearest minute and also minimum billing time is 10 minutes
+    run_minutes = max(10.0, math.ceil((end - start).total_seconds() / 60.0))
+    run_hours = run_minutes / 60.0
+    return run_hours
+
+def was_preempted(call_info):
+    # We treat Preempted and RetryableFailure the same.  The latter is a general case of the former
+    return call_info['executionStatus'] in ['Preempted', 'RetryableFailure']
+
+def calculate_cost(run_label,metadata, ignore_preempted, only_total_cost, print_header):
+    # set up pricing information
+    pricing = get_gce_pricing()
+    ssd_cost_per_gb_per_month = float(pricing["CP-COMPUTEENGINE-STORAGE-PD-SSD"])
+    ssd_cost_per_gb_hour = (ssd_cost_per_gb_per_month / (24 * 365 / 12))
+
+    hdd_cost_per_gb_per_month = float(pricing["CP-COMPUTEENGINE-STORAGE-PD-CAPACITY"])
+    hdd_cost_per_gb_hour = (hdd_cost_per_gb_per_month / (24 * 365 / 12))
+
+    disk_costs = {"PERSISTENT_SSD": ssd_cost_per_gb_hour, "PERSISTENT_HDD": hdd_cost_per_gb_hour}
+    
+    if print_header and not only_total_cost:
+        # print out a header
+        print(("\t".join(
+            ["run_id","task_name", "status", "machine_type", "total_hours", "cpu_cost_per_hour", "cpu_cost", "pe_total_hours",
+             "pe_cpu_cost_per_hour", "pe_cpu_cost", "failed_pe_total_hours", "failed_pe_cpu_cost", "disk_type",
+             "disk_size", "disk_gb_hours", "disk_cost", "failed_pe_ssd_gb_hours", "failed_pe_ssd_cost", "total_cost"])))
+    # iterate through the metadata file for each call
+    for k, v in metadata['calls'].items():
+        task_name = k
+
+        total_hours = 0
+        pe_total_hours = 0
+        failed_pe_total_hours = 0
+        machine_type = "unknown"
+        complete = True
+
+        for call_info in v:
+            # this is a subworkflow, recursively calculate cost on workflow metadata
+            if 'subWorkflowMetadata' in call_info:
+                calculate_cost(call_info['subWorkflowMetadata'], ignore_preempted, only_total_cost, False)
```

Comment:
This call to `calculate_cost` is missing a parameter

Reply:
good call. just fix it